src/main/scala/beam/router/RouteHistory.scala

```diff
@@ -73,4 +107,98 @@ class RouteHistory @Inject()() extends LazyLogging {
       }
     }
   }
+  override def notifyIterationEnds(event: IterationEndsEvent): Unit = {
+    val fileHeader = "timeBin,originTAZId,destTAZId,route"
+    val filePath = event.getServices.getControlerIO.getIterationFilename(
+      event.getServices.getIterationNumber,
+      RouteHistory.outputFileBaseName + ".csv.gz"
+    )
+
+    val flattenedRouteHistory: Iterable[(TimeBin, OriginTazId, DestTazId, String)] = routeHistory.flatMap {
```

Comment:
It seems to me that this might be read more clearly if it was in a for-comprehension, but maybe that's just my own preference. Take it or leave it.